### PR TITLE
fix: [OIP-592]: Add authentication UI to splunk connector

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/35-connectors/components/CreateConnector/SplunkConnector/CreateSplunkConnector.tsx [hunk 1]
```diff
@@ -38,20 +44,43 @@ function SplunkConfigStep(props: ConnectionConfigProps): JSX.Element {
...
   const { trackEvent } = useTelemetry()
 
+  const authOptions = getAuthOptions(getString)
+
   useTrackEvent(ConnectorActions.CreateConnectorLoad, {
     category: Category.CONNECTOR,
     connector_type: Connectors.Splunk
   })
 
+  const scope: ScopedObjectDTO | undefined = props.isEditMode
```
Comment (deepesh-ui) [inline src/modules/35-connectors/components/CreateConnector/SplunkConnector/CreateSplunkConnector.tsx:54]: isEditMode at desctructure at line 32

File: src/modules/35-connectors/components/CreateConnector/SplunkConnector/CreateSplunkConnector.tsx [hunk 2]
```diff
@@ -60,17 +89,40 @@ function SplunkConfigStep(props: ConnectionConfigProps): JSX.Element {
...
-              secretFieldValue={secretValue}
-              accountIdentifier={accountId}
-              projectIdentifier={projectIdentifier}
-              orgIdentifier={orgIdentifier}
-              onSuccessfulFetch={result => {
-                formikProps.setFieldValue('passwordRef', result)
-              }}
-            />
+
+            <Layout.Horizontal flex={{ justifyContent: 'space-between' }}>
```
Comment (deepesh-ui) [inline src/modules/35-connectors/components/CreateConnector/SplunkConnector/CreateSplunkConnector.tsx:93]: could we make this a  generic Auth Component  ?

Comment (praneshg239) [reply src/modules/35-connectors/components/CreateConnector/SplunkConnector/CreateSplunkConnector.tsx:93]: This cannot be made as a generic auth component, as the options differs for different connector types. Example, `Jenkins` connector has `No Auth` option where we do not have it in `Splunk` connector.

**Review by deepesh-ui — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/SplunkConnector/__tests__/CreateSplunkConnector.test.tsx [hunk 3]
```diff
@@ -24,11 +35,33 @@ const SplunkURL = 'https://splunk.com/api/v1/'
...
+  }
+}))
+
+jest.mock('@secrets/components/SecretInput/SecretInput', () => ({
+  ...(jest.requireActual('@secrets/components/SecretInput/SecretInput') as any),
+  default: function MockComponent() {
+    const { setFieldValue: formikSetValue } = useFormikContext()
+    useEffect(() => {
+      formikSetValue('tokenRef', {
+        identifier: 'cvng_appd_secret_atjPj0Innf',
```
Comment (reetika04) [inline src/modules/35-connectors/components/CreateConnector/SplunkConnector/__tests__/CreateSplunkConnector.test.tsx:57]: Please use dummy values

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]
UI css looks off:
1. No space between splunk icon and Overview bullet
2. Splunk Connector Details heading css is not correct.
3. Spacing is off for tab in screenshot.
4. Back/ Connect and Save buttons are not aligned and have wrong css. Please refer other connectors

**Comment by praneshg239** [comment]
> UI css looks off:
> 
> 1. No space between splunk icon and Overview bullet
> 2. Splunk Connector Details heading css is not correct.
> 3. Spacing is off for tab in screenshot.
> 4. Back/ Connect and Save buttons are not aligned and have wrong css. Please refer other connectors

image

File: src/modules/35-connectors/components/CreateConnector/SplunkConnector/__tests__/CreateSplunkConnector.test.tsx (hunk 3 above)
Comment (praneshg239) [reply src/modules/35-connectors/components/CreateConnector/SplunkConnector/__tests__/CreateSplunkConnector.test.tsx]: Done

**Review by deepesh-ui — APPROVED** [review]

**Review by meenakshi-harness — APPROVED** [review]

**Pull request merged**